onnxscript/function_libs/torch_aten/ops/nn.py

```diff
@@ -741,28 +741,104 @@ def aten_multilabel_margin_loss_forward(
     raise NotImplementedError()
 
 
+@torch_op("aten::nll_loss", trace_only=True)
 def aten_nll_loss(
-    self: TensorType,
-    target: TensorType,
-    weight: Optional[TensorType] = None,
+    self: TFloat,
+    target: Sequence[INT64],
+    weight: Optional[TFloat] = None,
     reduction: int = 1,
-    ignore_index: INT64 = -100,
-) -> TensorType:
-    """nll_loss(Tensor self, Tensor target, Tensor? weight=None, int reduction=Mean, SymInt ignore_index=-100) -> Tensor"""
+    ignore_index: int = -100,
+) -> TFloat:
+    # nll_loss(Tensor self, Tensor target, Tensor? weight=None, int reduction=Mean, SymInt ignore_index=-100) -> Tensor
+    reduction_vals = ["none", "mean", "sum"]
+    reduction_str = reduction_vals[reduction]
+
+    if weight is not None:
+        result = _aten_nll_loss_weight(
+            self, target, weight, reduction=reduction_str, ignore_index=ignore_index
+        )
+    else:
+        result = _aten_nll_loss_none(
+            self, target, reduction=reduction_str, ignore_index=ignore_index
+        )
+    return result
 
-    raise NotImplementedError()
 
+@torch_op("aten::nll_loss", overload=True)
+def _aten_nll_loss_weight(
+    self: TFloat,
+    target: Sequence[INT64],
```

Comment:
Is this supposed to be a sequence of ints (an attribute)? If so, the declaration should be `Sequence[int]`.

Reply:
the target should be a tensor with int type, so I changed it to TInt.